openmc/surface.py

```diff
@@ -0,0 +1,2261 @@
+from abc import ABCMeta, abstractmethod
+from collections import OrderedDict
+from copy import deepcopy
+from numbers import Real, Integral
+from xml.etree import ElementTree as ET
+from warnings import warn
+
+import numpy as np
+
+from openmc.checkvalue import check_type, check_value
+from openmc.region import Region, Intersection, Union
+from openmc.mixin import IDManagerMixin
+
+
+_BOUNDARY_TYPES = ['transmission', 'vacuum', 'reflective', 'periodic', 'white']
+
+_WARNING_UPPER = """\
+"{}(...) accepts an argument named '{}', not '{}'. Future versions of OpenMC \
+will not accept the capitalized version.\
+"""
+
+
+class Surface(IDManagerMixin, metaclass=ABCMeta):
```

Comment:
This is a good abstract class, but consider splitting up all of the subclasses of this class into their own submodules to limit the size of this file.